incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -109,10 +107,17 @@ func (r *HierarchyConfigReconciler) reconcile(ctx context.Context, log logr.Logg
 	origNS := nsInst.DeepCopy()
 
 	// Get singleton from apiserver. If it doesn't exist, initialize one.
-	inst, err := r.getSingleton(ctx, nm)
+	inst, deletingCRD, err := r.getSingleton(ctx, nm)
 	if err != nil {
 		return err
 	}
+	// If the CRD is being deleted, then there's no need to sync a missing
+	// CR. However, if the CR is _present_, we may need to update it to
+	// remove finalizers (see #824).
+	if deletingCRD && inst.CreationTimestamp.IsZero() {
```

Comment:
What does `CreationTimestamp.IsZero()` mean here?

Reply:
Exactly. Updated the comment.